From c444dc614936bd2083636b421ae1e4f5813b7a81 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Wed, 26 Dec 2018 02:21:31 +0100
Subject: [PATCH 1/1] efi_loader: signature of StartImage and Exit

We use u16* for Unicode strings and efi_uintn_t for UINTN. Correct the
signature of efi_exit() and efi_start_image().

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 include/efi_api.h             |  6 +++---
 lib/efi_loader/efi_boottime.c | 10 +++++-----
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/include/efi_api.h b/include/efi_api.h
index 1c4811f5657..1e2bbf60407 100644
--- a/include/efi_api.h
+++ b/include/efi_api.h
@@ -116,11 +116,11 @@ struct efi_boot_services {
 			struct efi_device_path *file_path, void *source_buffer,
 			efi_uintn_t source_size, efi_handle_t *image);
 	efi_status_t (EFIAPI *start_image)(efi_handle_t handle,
-					   unsigned long *exitdata_size,
-					   s16 **exitdata);
+					   efi_uintn_t *exitdata_size,
+					   u16 **exitdata);
 	efi_status_t (EFIAPI *exit)(efi_handle_t handle,
 				    efi_status_t exit_status,
-				    unsigned long exitdata_size, s16 *exitdata);
+				    efi_uintn_t exitdata_size, u16 *exitdata);
 	efi_status_t (EFIAPI *unload_image)(efi_handle_t image_handle);
 	efi_status_t (EFIAPI *exit_boot_services)(efi_handle_t, unsigned long);
 
diff --git a/lib/efi_loader/efi_boottime.c b/lib/efi_loader/efi_boottime.c
index 37f0f0e3380..92ba09a6828 100644
--- a/lib/efi_loader/efi_boottime.c
+++ b/lib/efi_loader/efi_boottime.c
@@ -1738,8 +1738,8 @@ error:
  * Return: status code
  */
 static efi_status_t EFIAPI efi_start_image(efi_handle_t image_handle,
-					   unsigned long *exit_data_size,
-					   s16 **exit_data)
+					   efi_uintn_t *exit_data_size,
+					   u16 **exit_data)
 {
 	struct efi_loaded_image_obj *image_obj =
 		(struct efi_loaded_image_obj *)image_handle;
@@ -1805,8 +1805,8 @@ static efi_status_t EFIAPI efi_start_image(efi_handle_t image_handle,
  */
 static efi_status_t EFIAPI efi_exit(efi_handle_t image_handle,
 				    efi_status_t exit_status,
-				    unsigned long exit_data_size,
-				    int16_t *exit_data)
+				    efi_uintn_t exit_data_size,
+				    u16 *exit_data)
 {
 	/*
 	 * TODO: We should call the unload procedure of the loaded
@@ -1815,7 +1815,7 @@ static efi_status_t EFIAPI efi_exit(efi_handle_t image_handle,
 	struct efi_loaded_image_obj *image_obj =
 		(struct efi_loaded_image_obj *)image_handle;
 
-	EFI_ENTRY("%p, %ld, %ld, %p", image_handle, exit_status,
+	EFI_ENTRY("%p, %ld, %zu, %p", image_handle, exit_status,
 		  exit_data_size, exit_data);
 
 	/* Make sure entry/exit counts for EFI world cross-overs match */
-- 
2.19.2

